src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,111 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int CAPACITY_MULTIPLIER = 2;
+    private Node<K, V>[] table;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        table = (Node<K,V>[]) new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getIndex(key);
+        Node<K, V> node = table[index];
+        if (node == null) {
+            table[index] = new Node<>(hash(key), key, value, null);
+            size++;
+        } else {
+            putOrSet(index, key, value);
+        }
+        increaseCapacity();
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> node = findNode(key);
+        return node != null ? node.value : null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void putOrSet(int index, K key, V value) {
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (node.isEqualKey(key)) {
+                node.setValue(value);
+                return;
+            }
+            node = node.next;
+        }
+        table[index] = new Node<>(hash(key), key, value, table[index]);
+        size++;
+    }
+
+    private Node<K, V> findNode(K key) {
```

Comment:
It's not critical, but you can use this logic in the getValue() method directly and avoid creating method, that is used only in that place